### PR TITLE
doc: Pass TermManager to Solver in Java examples

### PR DESCRIPTION
**Pull request by daniel-larraz**
It also replaces calls to deprecated functions of `Solver` with the corresponding function of `TermManager`.

### CODE REVIEW

File: examples/api/java/QuickStart.java [hunk 1]
```diff
@@ -20,19 +20,27 @@
 import java.util.Arrays;
 import java.util.List;
 
+import javax.naming.Context;
```
Comment (mudathirmahgoub) [inline examples/api/java/QuickStart.java:23]: You probably want to delete this line

File: examples/api/java/QuickStart.java [hunk 2]
```diff
@@ -41,26 +49,26 @@ public static void main(String args[]) throws CVC5ApiException
...
+      //! [docs-java-quickstart-3 start]
       solver.setLogic("ALL");
-      //! [docs-java-quickstart-2 end]
+      //! [docs-java-quickstart-3 end]
 
       // In this example, we will define constraints over reals and integers.
       // Hence, we first obtain the corresponding sorts.
-      //! [docs-java-quickstart-3 start]
+      //! [docs-java-quickstart-4 start]
       Sort realSort = solver.getRealSort();
```
Comment (mudathirmahgoub) [inline examples/api/java/QuickStart.java:59]:
```suggestion
      Sort realSort = tm.getRealSort();
```

File: examples/api/java/QuickStart.java [hunk 3]
```diff
@@ -41,26 +49,26 @@ public static void main(String args[]) throws CVC5ApiException
...
       Sort intSort = solver.getIntegerSort();
-      //! [docs-java-quickstart-3 end]
+      //! [docs-java-quickstart-4 end]
 
       // x and y will be real variables, while a and b will be integer variables.
       // Formally, their cpp type is Term,
       // and they are called "constants" in SMT jargon:
-      //! [docs-java-quickstart-4 start]
+      //! [docs-java-quickstart-5 start]
       Term x = solver.mkConst(realSort, "x");
```
Comment (mudathirmahgoub) [inline examples/api/java/QuickStart.java:67]:
```suggestion
      Term x = tm.mkConst(realSort, "x");
```

File: examples/api/java/QuickStart.java [hunk 4]
```diff
@@ -96,50 +104,50 @@ public static void main(String args[]) throws CVC5ApiException
...
       Term xVal = solver.getValue(x);
       Term yVal = solver.getValue(y);
-      //! [docs-java-quickstart-8 end]
+      //! [docs-java-quickstart-9 end]
 
       // It is also possible to get values for compound terms,
       // even if those did not appear in the original formula.
-      //! [docs-java-quickstart-9 start]
+      //! [docs-java-quickstart-10 start]
       Term xMinusY = solver.mkTerm(Kind.SUB, x, y);
```
Comment (mudathirmahgoub) [inline examples/api/java/QuickStart.java:131]:
```suggestion
      Term xMinusY = tm.mkTerm(Kind.SUB, x, y);
```

**Comment by mudathirmahgoub** [comment]
Thanks for updating the docs
I think you need to follow the cpp example (the code part) which I think the only one that was updated to use the term manager. 
https://cvc5.github.io/docs-ci/docs-main/api/cpp/quickstart.html
The python base api needs to be updated as well.

File: examples/api/java/QuickStart.java (hunk 1 above)
Comment (daniel-larraz) [reply examples/api/java/QuickStart.java]: This fixed the issue. Thanks!

**Comment by daniel-larraz** [comment]
> Thanks for updating the docs I think you need to follow the cpp example (the code part) which I think the only one that was updated to use the term manager. https://cvc5.github.io/docs-ci/docs-main/api/cpp/quickstart.html The python base api needs to be updated as well.

I noticed that the other Java examples also need to be updated. I will focus on updating the Java examples in this PR, and will open a new PR for Python.

**Review by mudathirmahgoub — APPROVED** [review]
LGTM

File: examples/api/java/QuickStart.java [hunk 5]
```diff
@@ -41,26 +47,26 @@ public static void main(String args[]) throws CVC5ApiException
...
+      //! [docs-java-quickstart-4 end]
 
       // x and y will be real variables, while a and b will be integer variables.
       // Formally, their cpp type is Term,
       // and they are called "constants" in SMT jargon:
-      //! [docs-java-quickstart-4 start]
-      Term x = solver.mkConst(realSort, "x");
+      //! [docs-java-quickstart-5 start]
+      Term x = tm.mkConst(realSort, "x");
       Term y = solver.mkConst(realSort, "y");
```
[the remark this answers is not in the document]
Comment (daniel-larraz) [reply examples/api/java/QuickStart.java]: I've already done it for QuickStart in commit e9ef5db. I will also do it for the other examples.

**Pull request merged**